modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-test/src/testIntegration/java/com/liferay/portal/reports/engine/console/service/test/EntryLocalServiceTest.java

```diff
@@ -0,0 +1,132 @@
+import com.liferay.mail.kernel.model.MailMessage;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.messaging.MessageListener;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.reports.engine.console.model.Definition;
+import com.liferay.portal.reports.engine.console.model.Entry;
+import com.liferay.portal.reports.engine.console.service.DefinitionLocalServiceUtil;
+import com.liferay.portal.reports.engine.console.service.EntryLocalServiceUtil;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.io.InputStream;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Alberto Sousa
+ */
+@RunWith(Arquillian.class)
+public class EntryLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testAddEntry() throws Exception {
+		Bundle bundle = FrameworkUtil.getBundle(EntryLocalServiceTest.class);
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
+		Dictionary<String, Object> dictionary = MapUtil.singletonDictionary(
+			"destination.name", DestinationNames.MAIL);
+
+		TestMessageListener testMessageListener = new TestMessageListener();
+
+		ServiceRegistration<?> serviceRegistration =
+			bundleContext.registerService(
+				MessageListener.class, testMessageListener, dictionary);
+
+		Definition definition = _addDefinition();
+
+		try {
+			Entry entry = EntryLocalServiceUtil.addEntry(
+				TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+				definition.getDefinitionId(), "txt", false, null, null, false,
+				StringPool.BLANK, StringPool.BLANK,
+				RandomTestUtil.randomString() + "@liferay.com",
+				StringPool.BLANK, StringPool.BLANK,
+				RandomTestUtil.randomString(), StringPool.BLANK,
+				ServiceContextTestUtil.getServiceContext());
+
+			for (FileAttachment fileAttachment :
+					testMessageListener.getFileAttachment()) {
+
+				Assert.assertFalse(FileUtil.exists(fileAttachment.getFile()));
+			}
+
+			EntryLocalServiceUtil.deleteEntry(entry);
```

Comment:
@abesousa do we need to delete the entry? please confirm if the data guard deletes it at the end of the test

Thank you!